frontend/src/pages/RenterCandidate.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useContext, useState } from "react";
+import { Helmet } from "react-helmet-async";
+import { useParams } from "react-router";
+import { Link, useNavigate } from "react-router-dom";
+import styled from "styled-components";
+import { Unit, approveUnit, getUnit, updateUnit } from "@/api/units";
+import { Page } from "@/components";
+import { NavBar } from "@/components/NavBar";
+import { Button } from "@/components/Button";
+import { getRenterCandidate, RenterCandidate } from "@/api/renter-candidates";
+
+export function RenterCandidate() {
+  const navigate = useNavigate();
+  const [renterCandidate, setRenterCandidate] = useState<RenterCandidate>();
+  const { id } = useParams();
+  const [loading, setLoading] = useState(true);
+
+  const fetchRenterCandidate = () => {
+    if (id !== undefined) {
+      setLoading(true);
+      void getRenterCandidate(id).then((result) => {
+        if (result.success) {
+          setRenterCandidate(result.data);
+        } else {
+          // Go back to the home page if the renter is not found
+          navigate("/");
+        }
+        setLoading(false);
+      });
+    }
+  };
+
+  React.useEffect(fetchRenterCandidate, []);
+
+  if (loading || !renterCandidate) {
+    // TODO: Loading state
+    return null;
+  }
+
+  const MainColumn = styled.div`
+    display: flex;
+    flex-direction: column;
+    background-color: #fbf7f3;
+    margin: 97px;
+    gap: 35px;
+  `;
+
+  const TopRow = styled.div`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+  `;
+
+  const Row = styled.div`
+    display: flex;
+    flex-direction: row;
+    gap: 20px;
+    align-items: center;
+  `;
+
+  const InfoRow = styled(Row)`
+    gap: 150px;
+    align-items: flex-start;
+  `;
+  const InfoColumn = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 15px;
+  `;
+
+  const Name = styled.p`
+    font-size: 40px;
+    font-family: "Neutraface Text";
+    font-weight: 700;
+  `;
+
+  const Id = styled.p`
+    font-size: 25px;
+    font-family: "Neutraface Text";
+    font-weight: 600;
+  `;
+
+  const InfoHeader = styled.p`
+    font-size: 18px;
+    font-family: "Montserrat";
+    font-weight: 600;
+  `;
+
+  const InfoText = styled(InfoHeader)`
+    font-weight: 400;
+  `;
+
+  const EditButton = styled(Button)`
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+  `;
```

Comment:
Can we move these outside of the function for [performance reasons](https://styled-components.com/docs/faqs#why-should-i-avoid-declaring-styled-components-in-the-render-method)?

Reply:
fixed!